src/providers/oauth-provider.js

```diff
@@ -122,7 +122,7 @@ function OAuthProvider() {
 
         data = queryString.stringify(data);
 
-        options = angular.extend({
+        options = angular.merge({
```

Comment:
Since you're only extending `options.headers` I would suggest to avoid deep copy and do the following:
```js
options.headers = angular.extend({
  'Authorization': undefined,
  'Content-Type': 'application/x-www-form-urlencoded'
}, options.headers);
```